### PR TITLE
[FIX] email_template_qweb: Decode body when encoded

### PR DESCRIPTION
**Pull request by pedrobaeza**
If the result is of type unicode, render method encodes it in utf-8.
We need to decode it in that case so that the rendering results correct.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please review @hbrunn @lasley @Yajo 

cc @Tecnativa

**Review by lasley — APPROVED** [review]
LGTM

Care to add a test covering this edge case?

**Review by hbrunn — APPROVED** [review]
yes, the extra test would be great

**Comment by pedrobaeza** [comment]
It's very difficult to test this, and more without having more dependencies (I detected it sending a sales order), so I prefer not to spend time on it if you don't mind.

**Comment by yajo** [comment]
But... I don't understand... What was exactly the problem with plain utf8? Aren't all fields stored as such?

**Comment by pedrobaeza** [comment]
No, they are passed directly to the wizard and accents appears incorrect

File: email_template_qweb/models/mail_template.py [hunk 1]
```diff
@@ -33,6 +33,10 @@ def generate_email(self, res_ids, fields=None):
                             'email_template': this,
                         })
                     )
+                    if isinstance(result[record_id]['body_html'], unicode):
```
Comment (yajo) [inline email_template_qweb/models/mail_template.py:38]: Please add a comment above this, explaining why it is needed

**Review by yajo — CHANGES_REQUESTED** [review]

**Review by yajo — APPROVED** [review]
I took the freedom to patch it myself. Let's merge :blush:
